cmd/clusterctl/pkg/client/repository/repository_local_unix_test.go

```diff
@@ -0,0 +1,388 @@
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/internal/test"
+)
+
+func Test_localRepository_newLocalRepository(t *testing.T) {
+
+	type fields struct {
+		provider              config.Provider
+		configVariablesClient config.VariablesClient
+	}
+	type want struct {
+		basepath       string
+		providerName   string
+		defaultVersion string
+		rootPath       string
+		componentsPath string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    want
+		wantErr bool
+	}{
+		{
+			name: "successfully creates new local repository object with a single version",
+			fields: fields{
+				provider:              config.NewProvider("provider-foo", "/base/path/provider-foo/v1.0.0/infrastructure-components.yaml", clusterctlv1.BootstrapProviderType),
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want: want{
+				basepath:       "/base/path",
+				providerName:   "provider-foo",
+				defaultVersion: "v1.0.0",
+				rootPath:       "",
+				componentsPath: "infrastructure-components.yaml",
+			},
+			wantErr: false,
+		},
+		{
+			name: "successfully creates new local repository object with a single version and no basepath",
+			fields: fields{
+				provider:              config.NewProvider("provider-foo", "/provider-foo/v1.0.0/infrastructure-components.yaml", clusterctlv1.BootstrapProviderType),
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want: want{
+				basepath:       "/",
+				providerName:   "provider-foo",
+				defaultVersion: "v1.0.0",
+				rootPath:       "",
+				componentsPath: "infrastructure-components.yaml",
+			},
+			wantErr: false,
+		},
+		{
+			name: "fails if an absolute path not specified",
+			fields: fields{
+				provider:              config.NewProvider("provider-foo", "./provider-foo/v1/infrastructure-components.yaml", clusterctlv1.BootstrapProviderType),
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want:    want{},
+			wantErr: true,
+		},
+		{
+			name: "fails if provider name does not match in the path",
+			fields: fields{
+				provider:              config.NewProvider("provider-foo", "/foo/bar/provider-bar/v1/infrastructure-components.yaml", clusterctlv1.BootstrapProviderType),
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want:    want{},
+			wantErr: true,
+		},
+		{
+			name: "fails if malformed path: invalid version directory",
+			fields: fields{
+				provider:              config.NewProvider("provider-foo", "/foo/bar/provider-foo/v.a.b.c/infrastructure-components.yaml", clusterctlv1.BootstrapProviderType),
+				configVariablesClient: test.NewFakeVariableClient(),
+			},
+			want:    want{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newLocalRepository(tt.fields.provider, tt.fields.configVariablesClient)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
This can be `t.Fatalf` without return (same in other tests)

Reply:
Done